pkg/ipam/node_manager.go

```diff
@@ -418,15 +418,14 @@ func (n *NodeManager) Get(nodeName string) *Node {
 // allocated or released, sorted by the number of addresses needed to be operated
 // in descending order. Number of addresses to be released is negative value
 // so that nodes with IP deficit are resolved first
+// The caller must hold the NodeManager lock
 func (n *NodeManager) GetNodesByIPWatermark() []*Node {
-	n.mutex.RLock()
```

Comment:
Why is it safe to drop this lock?

Reply:
This is only called from nodemanager.Resync: https://github.com/cilium/cilium/pull/26963/files#diff-28c460f7ee57d8ce6d40f9b55d3e627821a0e9158d7089e526fc75138b45096eR508

So the lock is being moved up to there.